hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1014,6 +1014,62 @@ ContainerCommandResponseProto handleWriteChunk(
     return getWriteChunkResponseSuccess(request, blockDataProto);
   }
 
+  /**
+   * Handle Write Chunk operation for closed container. Calls ChunkManager to process the request.
+   *
+   */
+  public void writeChunkForClosedContainer(ChunkInfo chunkInfo, BlockID blockID,
+                                           ChunkBuffer data, KeyValueContainer kvContainer)
+      throws IOException {
+
+    try {
+      Preconditions.checkNotNull(chunkInfo);
+      Preconditions.checkNotNull(data);
+      long writeChunkStartTime = Time.monotonicNowNanos();
+      checkContainerClose(kvContainer);
+
+      DispatcherContext dispatcherContext = DispatcherContext.getHandleWriteChunk();
+      chunkManager.writeChunk(kvContainer, blockID, chunkInfo, data,
+          dispatcherContext);
+
+      // Increment write stats for WriteChunk after write.
+      metrics.incContainerBytesStats(Type.WriteChunk, chunkInfo.getLen());
+      metrics.incContainerOpsLatencies(Type.WriteChunk, Time.monotonicNowNanos() - writeChunkStartTime);
```

Comment:
Might be cleaner to use or introduce something similar to org.apache.hadoop.ozone.util.MetricUtil#captureLatencyNs(org.apache.hadoop.metrics2.lib.MutableRate, org.apache.ratis.util.function.CheckedSupplier<T,E>) to capture latency.

Reply:
We can do it, but I'm reusing the same previous implementation to avoid code duplication.